### ISSUE TITLE
CSS Image is server-rendered as [object object] until JS loads to replace it

### ISSUE THREAD

**Issue by josephdburdick** [sysgears/apollo-fullstack-starter-kit#51]
Hello, I've been facing issues with leveraging image assets in this project and, while I could stand to understand Webpack better, it would be nice to see some guidance in the documentation as to where I can learn more about the following (within the _context_ of this project's setup):

1. Add an inline image
2. Add an SVG
3. Load a CSS background image

As it stands I cannot render a background image without JS turned on. I load a png as a background-image in Sass and it is defined as [object object] until Javascript renders and replaces it. I think that the SSR should handle this without JS being enabled, it will make pages render faster.

Thanks very much!

**Comment by vlasenko** [comment]
Hi,

I'm not sure what you are talking about, this tickets need details, how to reproduce each of the issue you are having. Because I don't have any problems with images, everything works as expected.

**Comment by josephdburdick** [comment]
No problem @vlasenko, hopefully the following will suffice.
Basically I have a div with a class of `.brief`. Inside of `.brief` I have the following defined: 
```
.brief {
    color: white;
    position: relative;
    background: $violet url('../../assets/images/bg/bg_brand-blocks.png');
    background-position: center;
    background-repeat: no-repeat;
    background-size: contain;
  }
```
When I turn off Javascript in the browser (rendering pure SSR only) the image background png doesn't load. Inspecting the element looks like this:
<img width="316" alt="screen shot 2017-02-26 at 2 27 31 pm" src="https://cloud.githubusercontent.com/assets/1349077/23341955/eec7aa62-fc2f-11e6-8fb2-92a459e646f3.png">

When I turn Javascript back on the image renders in the browser and inspector results in this:
<img width="654" alt="screen shot 2017-02-26 at 2 28 08 pm" src="https://cloud.githubusercontent.com/assets/1349077/23341956/f1619378-fc2f-11e6-9148-e7cdb274f88b.png">

The behavior for an SVG background-image is a little different. It never loads. 
With or without Javascript enabled:
<img width="480" alt="screen shot 2017-02-26 at 2 31 26 pm" src="https://cloud.githubusercontent.com/assets/1349077/23341978/a9cc2a68-fc30-11e6-85ac-f3a800dbb96e.png">

Opening this url leads me here:
<img width="1063" alt="screen shot 2017-02-26 at 2 31 43 pm" src="https://cloud.githubusercontent.com/assets/1349077/23341979/a9d45fe4-fc30-11e6-8af9-2d20e0be075b.png">

Just to be even more thorough the following is my webpack config, it shouldn't be any different from this project:
```
// Don't use ES6 here to stay compatible with ESLint plugins

const webpack = require('webpack');
const ManifestPlugin = require('webpack-manifest-plugin');
const ExtractTextPlugin = require('extract-text-webpack-plugin');
const merge = require('webpack-merge');
const nodeExternals = require('webpack-node-externals');
const path = require('path');
const pkg = require('../package.json');
const _ = require('lodash');

global.__DEV__ = process.argv.length >= 3 && process.argv[2] === 'watch';
const buildNodeEnv = __DEV__ ? 'development' : 'production';

let basePlugins = [];

if (__DEV__) {
  basePlugins.push(new webpack.NamedModulesPlugin());
} else {
  basePlugins.push(new webpack.optimize.UglifyJsPlugin({ minimize: true }));
}

const baseConfig = {
  module: {
    noParse: [],
    loaders: [
      {
        test: /\.jsx?$/,
        loader: 'babel',
        exclude: /(node_modules|bower_components)/,
        query: {
          cacheDirectory: __DEV__,
          presets: ['es2015', 'es2017', 'react'],
          plugins: ['transform-runtime', 'transform-decorators-legacy', 'transform-class-properties'],
        },
      },
      { test: /\.json$/, loader: 'json' },
      { test: /\.graphqls/, loader: 'raw' },
      {
        test: /\.(graphql|gql)$/,
        exclude: /node_modules/,
        loader: 'graphql-tag/loader'
      },
      { test: /\.woff(2)?(\?v=[0-9]\.[0-9]\.[0-9])?$/, loader: "url-loader?limit=10000&minetype=application/font-woff" },
      { test: /\.(ttf|eot|svg)(\?v=[0-9]\.[0-9]\.[0-9])?$/, loader: "file-loader" }
    ]
  },
  resolve: {
    root: path.resolve(__dirname, '../src'),
    modulesDirectories: ['node_modules'],
    extensions: ['', '.js', '.jsx']
  },
  plugins: basePlugins,
  bail: !__DEV__
};

let serverPlugins = [
  new webpack.BannerPlugin('require("source-map-support").install();',
      { raw: true, entryOnly: false }),
  new webpack.DefinePlugin(Object.assign({__CLIENT__: false, __SERVER__: true, __SSR__: pkg.app.ssr,
    __DEV__: __DEV__, 'process.env.NODE_ENV': `"${buildNodeEnv}"`}))
];

const serverConfig = merge.smart(_.cloneDeep(baseConfig), {
  devtool: __DEV__ ? '#cheap-module-source-map' : '#source-map',
  target: 'node',
  entry: {
    index: ['babel-polyfill', './src/server/index.js']
  },
  node: {
    __dirname: true,
    __filename: true
  },
  externals: nodeExternals({
    whitelist: [/^webpack/]
  }),
  module: {
    loaders: [
      {
        test: /\.scss$/,
        loaders: __DEV__ ? [
          'isomorphic-style-loader',
          'css',
          'postcss',
          'sass'] : ['ignore-loader']
      },
      { test: /\.(woff2?|svg|png|ico|jpg|ttf|eot)$/, loader: 'ignore-loader' }
    ]
  },
  output: {
    devtoolModuleFilenameTemplate: __DEV__ ? '../../[resource-path]' : undefined,
    devtoolFallbackModuleFilenameTemplate: __DEV__ ? '../../[resource-path];[hash]' : undefined,
    filename: '[name].js',
    sourceMapFilename: '[name].[chunkhash].js.map',
    path: pkg.app.backendBuildDir,
    publicPath: '/'
  },
  plugins: serverPlugins
});

let clientPlugins = [
  new ManifestPlugin({
    fileName: 'assets.json'
  }),
  new webpack.DefinePlugin(Object.assign({__CLIENT__: true, __SERVER__: false, __SSR__: pkg.app.ssr,
    __DEV__: __DEV__, 'process.env.NODE_ENV': `"${buildNodeEnv}"`})),
];

if (!__DEV__) {
  clientPlugins.push(new ExtractTextPlugin('[name].[contenthash].css', { allChunks: true }));
  clientPlugins.push(new webpack.optimize.CommonsChunkPlugin(
    "vendor",
    "[name].[hash].js",
    function (module) {
      return module.resource && module.resource.indexOf(path.resolve('./node_modules')) === 0;
    }
  ));
}

const clientConfig = merge.smart(_.cloneDeep(baseConfig), {
  devtool: __DEV__ ? '#eval' : '#source-map',
  entry: {
    bundle: ['babel-polyfill', './src/client/index.jsx']
  },
  module: {
    loaders: [
      {
        test: /\.scss$/,
        loader: __DEV__ ? 'style!css?importLoaders=1!postcss?sourceMap=inline!sass' : ExtractTextPlugin.extract("style", "css!postcss!sass")
      },
      { test: /\.(woff2?|svg|png|ico|jpg|xml)$/, loader: 'url?name=[hash].[ext]&limit=10000' },
      { test: /\.(ttf|eot)$/, loader: 'file?name=[hash].[ext]' },
    ]
  },
  output: {
    filename: '[name].[hash].js',
    path: pkg.app.frontendBuildDir,
    publicPath: '/'
  },
  plugins: clientPlugins
});

const dllConfig = merge.smart(_.cloneDeep(baseConfig), {
  entry: {
    vendor: _.keys(pkg.dependencies),
  },
  plugins: [
    new webpack.DllPlugin({
      name: '[name]',
      path: path.join(pkg.app.frontendBuildDir, '[name]_dll.json'),
    }),
  ],
  output: {
    filename: '[name].[hash]_dll.js',
    path: pkg.app.frontendBuildDir,
    library: '[name]',
  },
});

module.exports =

  process.argv.length >= 2 && process.argv[1].indexOf('mocha-webpack') >= 0 ?
    serverConfig :
    [ serverConfig, clientConfig, dllConfig ];
```
If there's any other information you need please let me know, thank you for looking into this!

**Comment by vlasenko** [comment]
Yes, you are right. I have a problem in backend webpack config. Frontend and backend should use the SAME config for images. It was the same from the beginning, but at some stage, I just started to use `ignore-loader` on backend for images. It was a mistake. I have returned back the same config for images for backend and frontend. See the commit above. If you have more questions about images, lets discuss.
